[train+tune][doc] Remove docs sections recommending `RAY_AIR_LOCAL_CACHE_DIR` (#44284)
ray-project/ray: doc/source/train/user-guides/persistent-storage.rst

            ...

.. _train-storage-advanced:

Advanced configuration
----------------------

.. _train-local-staging-dir:

Setting the local staging directory
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. warning::

    Prior to 2.10, the ``RAY_AIR_LOCAL_CACHE_DIR`` environment variable and ``RunConfig(local_dir)``
    were ways to configure the local staging directory to be outside of the home directory (``~/ray_results``).

>     **These configurations no longer used to configure the local staging directory.

Reviewer: ```suggestion
    **These configurations are no longer used to configure the local staging directory.
```

>     Please instead use ``RunConfig(storage_path)`` to configure where your

Reviewer: Rendering gets messed up here, so we can either unbold this line, or do this:
```suggestion
    Please instead use** ``RunConfig(storage_path)`` **to configure where your
```

Advanced configuration
----------------------

Setting the local staging directory
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Apart from files written to the ``storage_path``, Ray Train also writes to some
logfiles and metadata files in a *local staging directory* before they get
persisted (copied/uploaded) to the ``storage_path``.

> By default, this local staging directory is a sub-directory of the temporary Ray session

Reviewer: Add some more detail here what sub-directory the user should look at.

> directory (``/tmp/ray/session_latest``), which is also where Ray Core logs are dumped.

Reviewer: Not specific to logs since that would be in the `/logs` sub-directory.
```suggestion
directory (``/tmp/ray/session_latest``), which is also where other temporary Ray files are dumped.
```

> Customize the location of the staging directory by customizing the location of the
> temporary Ray session directory. See :ref:`temp-dir-log-files` for how to set this directory.

Reviewer: Can make this more concise.

```suggestion
Customize the location of the staging directory by :ref:`customizing the location of the
temporary Ray session directory <temp-dir-log-files>`.
```

.. warning::

    Prior to 2.10, the ``RAY_AIR_LOCAL_CACHE_DIR`` environment variable was the way to configure
    the local staging directory to be outside of the home directory (``~/ray_results``).

    ``RunConfig(local_dir)`` was another deprecated alternative to set the local staging directory.

    These options are now deprecated, and you should instead set ``storage_path``.
    When ``storage_path`` is set, there will no longer be any files written to

>     ``~/ray_results`` starting from Ray 2.10.

Reviewer: This is a bit confusing since there's a bunch of branching logic based on env var / storage path. Could we maybe simplify this to just focus on the staging directory being set by `RAY_AIR_LOCAL_CACHE_DIR` or `local_dir`? And not talk about `storage_path`?

Author: I kept the mention about switching to storage path as a migration path, because users were setting ONLY the environment variable. But now that the env variable doesn't do anything, they need to switch to setting storage path.
